tests/RobotFramework/tests/cumulocity/registration/registration_livecycle.robot

```diff
@@ -30,6 +30,19 @@ Child device registration
     Cumulocity.Set Device    ${DEVICE_SN}
     Cumulocity.Should Be A Child Device Of Device    ${CHILD_SN}
 
+    # Deregister Child device
+    Execute Command    mosquitto_sub --remove-retained -W 3 -t "te/device/${CHILD_SN}/+/+/#"    exp_exit_code=27
+    
+    # Check if deregistration was successful
+    ${retained_message_count}=    Get Retained Message Count    te/device/${CHILD_SN}/+/+/#
+    Should Be Equal As Numbers    ${retained_message_count}    0    
+
+    # Checking if child device will be recreated after mapper restart
+    Restart Service    tedge-mapper-c8y    
+    Sleep    5s    reason=Allowing startup to complete
+    ${retained_message_count}=    Get Retained Message Count    te/device/${CHILD_SN}/+/+/#
+    Should Be Equal As Numbers    ${retained_message_count}    0 
```

Comment:
A better validation would be to try and recreate the device by publishing the registration message again and validate that the mapper emits the mapped 101 message.

Reply:
it was @reubenmiller choice to go this way, if he agrees....

Reply:
This validation makes sure that the mapper doesn't re-publish the deleted entity back to the broker, which it was never doing. But it's still fine as an external backbox test.

But the key thing that we want to make sure is the fact that the mapper doesn't have any stale references to the deleted entity in its in-memory entity store after the restart. The only available way to check that currently is to send another registration message and make sure that the mapper doesn't ignore it as duplicate, but really maps that message.

Reply:
Once we have an official deregister api, then we can improve the assertions to check if the entities are registered in the cloud or not after deletion. We didn't want to invest too much into the assertions yet as it would of involved manually deleting the cloud entity via Cumulocity API (the part not yet implemented by thin-edge.io).

Reply:
Technically, we don't need to delete the device from the cloud for this specific assertion, but just make sure that the mapper emits the 101 message for the device that is recreated. Obviously, that 101 message would result in a "device already exists" error on C8y, but that can be ignored. That 101 message getting generated is all that we care about.
 
That being said, since the assertion might be different with the official deregister API (although the above assertion would still be valid), we can skip that if this effort is going to be wasted.

Reply:
Though I would prefer if the tests are decoupled from the implementation (e.g. it does not have to know anything about the c8y SmartREST messages), so we'll just skip it for now.